src/Illuminate/Auth/Passwords/PasswordBroker.php

```diff
@@ -200,43 +213,72 @@ public function getUser(array $credentials)
      * Create a new password reset token for the given user.
      *
      * @param  CanResetPasswordContract $user
+     * @param  int $expiration
      * @return string
      */
-    public function createToken(CanResetPasswordContract $user)
+    public function createToken(CanResetPasswordContract $user, $expiration)
     {
-        return $this->tokens->create($user);
+        $payload = $this->buildPayload($user, $user->getEmailForPasswordReset(), $expiration);
+
+        return hash_hmac('sha256', $payload, $this->getKey());
     }
 
     /**
-     * Delete the given password reset token.
+     * Validate the given password reset token.
      *
-     * @param  string  $token
-     * @return void
+     * @param  CanResetPasswordContract $user
+     * @param  array $credentials
+     * @return bool
      */
-    public function deleteToken($token)
+    public function validateToken(CanResetPasswordContract $user, array $credentials)
     {
-        $this->tokens->delete($token);
+        $payload = $this->buildPayload($user, $credentials['email'], $credentials['expiration']);
+
+        return hash_equals($credentials['token'], hash_hmac('sha256', $payload, $this->getKey()));
     }
 
     /**
-     * Validate the given password reset token.
+     * Validate the given expiration timestamp.
      *
-     * @param  CanResetPasswordContract $user
-     * @param  string $token
+     * @param  int $expiration
      * @return bool
      */
-    public function tokenExists(CanResetPasswordContract $user, $token)
+    public function validateTimestamp($expiration)
+    {
+        return Carbon::createFromTimestamp($expiration)->isFuture();
+    }
+
+    /**
+     * Return the application key.
+     *
+     * @return string
+     */
+    public function getKey()
     {
-        return $this->tokens->exists($user, $token);
+        $key = $this->app['config']['app.key'];
+
+        if (Str::startsWith($key, 'base64:')) {
```

Comment:
Ah, well that kind of code duplication is getting a bit messy. Going off topic a bit but I've proposed some improvements to this laravel/internals/issues/416